src/protocol/command.c

```diff
@@ -54,61 +48,56 @@ protocolCommandNew(const StringId command)
 }
 
 /**********************************************************************************************************************************/
-ProtocolCommand *
-protocolCommandParamAdd(ProtocolCommand *this, const Variant *param)
+void
+protocolCommandPut(ProtocolCommand *const this, IoWrite *const write)
 {
     FUNCTION_TEST_BEGIN();
         FUNCTION_TEST_PARAM(PROTOCOL_COMMAND, this);
-        FUNCTION_TEST_PARAM(VARIANT, param);
     FUNCTION_TEST_END();
 
     ASSERT(this != NULL);
+    ASSERT(write != NULL);
 
-    MEM_CONTEXT_BEGIN(this->memContext)
-    {
-        // Create parameter list if not already created
-        if (this->parameterList == NULL)
-            this->parameterList = varNewVarLst(varLstNew());
+    // Write the command and flush to be sure the command gets sent immediately
+    PackWrite *commandPack = pckWriteNew(write);
+    pckWriteU32P(commandPack, protocolMessageTypeCommand, .defaultWrite = true);
```

Comment:
Why is .defaultWrite = true needed here"? The value of protocolMessageTypeCommand = 2  (non-zero) so it will always be written, no?

Reply:
If we change the enum then 0 might not get written. I'd rather it were explicit that the value will be written.